src/matches.rs

```diff
@@ -112,34 +112,40 @@ impl LateLintPass for MatchPass {
 fn check_single_match(cx: &LateContext, ex: &Expr, arms: &[Arm], expr: &Expr) {
     if arms.len() == 2 &&
        arms[0].pats.len() == 1 && arms[0].guard.is_none() &&
-       arms[1].pats.len() == 1 && arms[1].guard.is_none() &&
-       is_unit_expr(&arms[1].body) {
+       arms[1].pats.len() == 1 && arms[1].guard.is_none() {
```

Comment:
if the match arm is a (multiline) block, the `if let` version takes up one line less:

```rust
match x {
    Y => println!("Jupiter"),
    _ => {
        println!("Mars");
        println!("Venus");
    },
}
```

vs

```rust

if let Y = x {
    println!("Jupiter");
} else {
    println!("Mars");
    println!("Venus");
}